Iterate over points in tree

clbarnes opened the pull request:
Resolves #2

clbarnes committed:
Iterate over points in tree

stephanemagnenat commented:
Thank you for this PR. Overall it looks good.

There is one thing I'm not fully convinced, and that is the name of the function: `iter`. We are iterating the points in the tree, and not the tree itself. For the latter, I would assume to receive nodes and not points from the iterator.

What about naming the function `iter_points` for better clarity? If you agree, could you change the function in your PR to that naming? Then I can merge.

clbarnes committed:
More iteration methods, Point construction

Separate methods for just iterating through points,
and provide methods for constructing P: Point from slices.

clbarnes commented:
Sure! I've split the point iteration out from the idx + point iteration, and also added a couple of convenience methods to Point for constructing them from a slice.

stephanemagnenat commented:
Thank you for your contribution!
